### PR TITLE
Support for Dark Mode

### LINKED ISSUE #1450: Support for Dark Mode Stage II. remove StyleManager
Second step of the implementation of support for Dark Mode.

As discussed in p99K0U-1Z4-p2, we split the implementation in two stages. First, adding support for Muriel colours, and second, replacing the StyleManager.

This store is about removing the StyleManager, and making sure that the new colour palette is applied.

For more info:
p99K0U-1Vn-p2
p99K0U-1Xr-p2

### PR DESCRIPTION
**Pull request by ctarda**
Fixes #1450 

This PR implements support for Dark Mode.

| Light Mode | Dark Mode |
| ----------- | ----------- |
|  |  |
|  |  |
|  |  |
|  |  |
|  |  |
|  |  |

## Changes
- Remove the preexisting interface abstracting the Style Manager
- Consume colours declared in `UIColor+SemanticColors`

## Testing
- Download the build associated to the PR
- Run on device. 
- Navigate the app
- Take screenshots of what is not right, and annotate indicating the right semantic color that needs to be used

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ctarda** [comment]
Known issues: 
- The in login prologue needs to be split in two (logo and WooCommerce literal)

File: WooCommerce/Classes/Extensions/Ghost+Woo.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import WordPressUI
+
+/// Style for ghosting that does not show calles with a white background.
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/Ghost+Woo.swift]:
```suggestion
/// Style for ghosting that will not show calls with a white background.
```

**Comment by mindgraffiti** [comment]
Another way to test, using the simulator:
- go to the Settings App > Developer > turn on Dark Appearance
- build & run

File: WooCommerce/Classes/Authentication/Prologue/LoginPrologueViewController.swift [hunk 2]
```diff
@@ -108,7 +114,7 @@ private extension LoginPrologueViewController {
         let title = NSLocalizedString("Log in with Jetpack", comment: "Authentication Login Button")
         loginButton.titleLabel?.adjustsFontForContentSizeCategory = true
         loginButton.setTitle(title, for: .normal)
-        loginButton.setTitleColor(StyleManager.wooSecondary, for: .normal)
+        loginButton.setTitleColor(.black, for: .normal)
```
[the remark this answers is not in the document]
Comment (ctarda) [reply WooCommerce/Classes/Authentication/Prologue/LoginPrologueViewController.swift]: We should, but as far as I can tell there is no semantic color that is black (or similar) on the three modes we have to support (pre-ios 13, light and dark)

File: WooCommerce/Classes/Extensions/UIImage+Dot.swift [hunk 3]
```diff
@@ -12,7 +12,7 @@ extension UIImage {
     func imageWithTopRightDot(imageOrigin: CGPoint,
                               finalSize: CGSize,
                               dotDiameter: CGFloat = 7,
-                              dotColor: UIColor = StyleManager.announcementDotColor) -> UIImage? {
+                              dotColor: UIColor = .red) -> UIImage? {
```
[the remark this answers is not in the document]
Comment (ctarda) [reply WooCommerce/Classes/Extensions/UIImage+Dot.swift]: Yes, it should probably be .error

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Notes Section/Add Order Note/NewNoteViewController.swift [hunk 4]
```diff
@@ -128,7 +128,7 @@ private extension NewNoteViewController {
         }
 
         cell.iconImage = .asideImage
-        cell.iconTint = isCustomerNote ? StyleManager.statusPrimaryBoldColor : StyleManager.wooGreyMid
+        cell.iconTint = isCustomerNote ? .blue : .textSubtle
```
[the remark this answers is not in the document]
Comment (ctarda) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Notes Section/Add Order Note/NewNoteViewController.swift]: This one is supposed to be a semantic one (as declared in `UIColor+SemanticColors`). I'll double check it gets hit

File: WooCommerce/Classes/ViewRelated/Products/View Models/ProductsTabProductViewModel.swift [hunk 5]
```diff
@@ -26,11 +26,11 @@ private extension Product {
 
         let attributedString = NSMutableAttributedString(string: detailsText,
                                                          attributes: [
-                                                            .foregroundColor: StyleManager.wooGreyMid,
+                                                            .foregroundColor: UIColor.textSubtle,
                                                             .font: StyleManager.footerLabelFont
             ])
         if let statusText = statusText {
-            attributedString.addAttributes([.foregroundColor: StyleManager.highlightTextColor],
+            attributedString.addAttributes([.foregroundColor: UIColor.orange],
```
[the remark this answers is not in the document]
Comment (ctarda) [reply WooCommerce/Classes/ViewRelated/Products/View Models/ProductsTabProductViewModel.swift]: This one is declared by us as well

**Comment by ctarda** [comment]
Thanks folks!

I've pushed a bunch of commits to update:

- tab bar selected icon tint
- orders badge border
- Review stars (white background)
- Text color of sheet with order statuses
- Payment Details (white background)
- Accesory icon in Order Status list
- Switch, icon and background of text view in New Note screen
- Buttons in review details (still looks bad?)

**Comment by astralbodies** [comment]
I'm taking over the authoring of the PR as @ctarda has his hands full with the new bits. 😄

**Comment by astralbodies** [comment]
Just checked in fixes for the login prologue. @kyleaparker did a quick review and they looked good.

Simulator Screen Shot - iPhone 11 Pro Max - 2019-11-19 at 09 58 06

File: WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsCommentTableViewCell.swift [hunk 6]
```diff
@@ -315,7 +324,7 @@ private struct Approve {
 private struct Star {
     static let size        = Double(18)
     static let filledImage = UIImage.starImage(size: Star.size,
-                                               tintColor: StyleManager.goldStarColor)
+                                               tintColor: .warning)
     static let emptyImage = UIImage.starImage(size: Star.size,
-                                              tintColor: StyleManager.wooGreyLight)
+                                              tintColor: UIColor.listIcon)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsCommentTableViewCell.swift]: nit: `UIColor` probably not needed

File: WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsV4PeriodViewController.swift [hunk 7]
```diff
@@ -242,11 +243,14 @@ private extension StoreStatsV4PeriodViewController {
...
-        borderView.backgroundColor = StyleManager.wooGreyBorder
+        view.backgroundColor = .listForeground
+        containerStackView.backgroundColor = .listForeground
+        timeRangeBarView.backgroundColor = .listForeground
+        visitorsStackView.backgroundColor = .listForeground
+        borderView.backgroundColor = .listSmallIcon
 
         // Time range bar bottom border view
-        timeRangeBarBottomBorderView.backgroundColor = StyleManager.wooGreyBorder
+        timeRangeBarBottomBorderView.backgroundColor = .listForeground
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsV4PeriodViewController.swift]: I've started using `.divider` for border views, not sure if we should use `.divider` or `.listForeground` for the borders?

File: RELEASE-NOTES.txt [hunk 8]
```diff
@@ -9,6 +9,8 @@
 - bugfix: fixed UI appearance on cells of Order List when tapping with dark mode enabled.
 - bugfix: Reviews no longer convert to partial dark mode. Dark mode coming soon!
 - bugfix: Order Details now has the right space between cells.
+- Enhancement: Support for dark mode
+- bugfix: Settings no longer convert to partial dark model.
```
Comment (jaclync) [inline RELEASE-NOTES.txt]: could probably move these under release 3.2?

Comment (jaclync) [reply RELEASE-NOTES.txt]: not sure what "Settings no longer convert to partial dark model" means, does it still apply?

**Pull request merged**